src/controller/java/generated/java/matter/devicecontroller/cluster/clusters/ContentAppObserverCluster.kt

```diff
@@ -39,73 +48,290 @@ class ContentAppObserverCluster(
     encodingHint: String,
     timedInvokeTimeoutMs: Int? = null
   ): ContentAppMessageResponse {
-    val commandId = 0L
+    val commandId: UInt = 0u
+    val timeoutMs: Duration =
+      timedInvokeTimeoutMs?.let { Duration.ofMillis(it.toLong()) } ?: Duration.ZERO
 
-    if (timedInvokeTimeoutMs != null) {
-      // Do the action with timedInvokeTimeoutMs
-    } else {
-      // Do the action without timedInvokeTimeoutMs
-    }
+    val tlvWriter = TlvWriter()
+    tlvWriter.startStructure(AnonymousTag)
+
+    val TAG_DATA_REQ: Int = 0
+    data?.let { tlvWriter.put(ContextSpecificTag(TAG_DATA_REQ), data) }
+
+    val TAG_ENCODING_HINT_REQ: Int = 1
+    tlvWriter.put(ContextSpecificTag(TAG_ENCODING_HINT_REQ), encodingHint)
+    tlvWriter.endStructure()
+
+    val request: InvokeRequest =
+      InvokeRequest(
+        CommandPath(endpointId, clusterId = CLUSTER_ID, commandId),
+        tlvPayload = tlvWriter.getEncoded(),
+        timedRequest = timeoutMs
+      )
+
+    val response: InvokeResponse = controller.invoke(request)
+    logger.log(Level.INFO, "Invoke command succeeded: ${response}")
+
+    val tlvReader = TlvReader(response.payload)
```

Comment:
Is ordering of content guaranteed by the spec?

I thought all TLV data can be random order, in which case we need to process using loops and keep track if we got data or not.

Reply:
Good catch, currently we don't have problem since both TLV and Kotlin are driven from the same IDL files, so they are in order, but this is not guaranteed by the spec.

Reply:
Updated using loops and keep track if we got data or not.